ci/checks/doxygen.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Copyright (c) 2022-2024, NVIDIA CORPORATION.
+###############################
+# cuDF doxygen warnings check #
+###############################
+
+# skip if doxygen is not installed
+if ! [ -x "$(command -v doxygen)" ]; then
+  echo -e "warning: doxygen is not installed"
+  exit 0
+fi
+
+# Utility to return version as number for comparison
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+# doxygen supported version 1.9.1
+DOXYGEN_VERSION=`doxygen --version`
+if [ ! $(version "$DOXYGEN_VERSION") -eq $(version "1.9.1") ] ; then
+  echo -e "warning: Unsupported doxygen version $DOXYGEN_VERSION"
+  echo -e "Expecting doxygen version 1.9.1"
```

Comment:
Since the version being pulled in CI is 1.10.0:


```suggestion
# doxygen supported version 1.10.0
DOXYGEN_VERSION=`doxygen --version`
if [ ! $(version "$DOXYGEN_VERSION") -eq $(version "1.10.0") ] ; then
  echo -e "warning: Unsupported doxygen version $DOXYGEN_VERSION"
  echo -e "Expecting doxygen version 1.10.0"
```